bin/oneoff/backfill_data/school_info_id

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env ruby
+
+# This script copies user_id-school_info_id pairs from the pd_enrollments
+# table into the school_info_id column in users.
+
+require_relative '../../../dashboard/config/environment'
+
+# build hash mapping school_info_id to school_id
+school_info_id_school_id_mapping = {}
+
+batch_number = 0
+SchoolInfo.find_in_batches(batch_size: 2500) do |batch|
+  puts "PROCESSING SCHOOL_INFO: #{batch_number}..."
+  batch.each do |school_info|
+    school_info_id_school_id_mapping[school_info.id] = school_info.school_id
+  end
+  puts "PROCESSED SCHOOL_INFO: #{batch_number}..."
+  batch_number += 1
+end
+
+batch_number = 0
+User.find_in_batches(batch_size: 2_500) do |batch|
+  puts "PROCESSING USERS: #{batch_number}..."
+  ActiveRecord::Base.transaction do
+    batch.each do |user|
+      # only do this if user has no school_info_id or their school_info_id is
+      # not associated with a school_id
+      # (don't do it if the user already has a school_info_id associated with a school_id)
+      if school_info_id_school_id_mapping[user.school_info_id]
+        next
+      end
+
+      enrollments = Pd::Enrollment.for_user(user)
+      if enrollments.empty?
+        next
+      end
+      enrollments_with_school_id = enrollments.select {|e| school_info_id_school_id_mapping[e.school_info_id]}
+
+      # if the user has pd_enrollments whose school_info_id is associated with
+      # a school_id, update the user with the most recent of those pd_enrollments' school_info_id
+      if !enrollments_with_school_id.empty?
+        enrollment = enrollments_with_school_id.sort_by {|e| e[:updated_at]}.last
+        user.update(school_info_id: enrollment.school_info_id)
+      # if user doesn't have a school_info_id at all, update with most recent
+      # enrollment school_info_id even though it doesn't have school_id
+      elsif user.school_info_id.blank?
```

Comment:
nit: isn't this an id reference? It won't ever be empty string, it's either nil or a valid id, so instead of `blank?` just check `nil?`